components/toggle/Toggle.js

```diff
@@ -1,28 +1,62 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import { t } from 'ttag';
+import { c } from 'ttag';
 
 import Input from '../input/Input';
-import Label from '../label/Label';
+import Icon from '../icon/Icon';
+
+const label = (key) => {
+    const I18N = {
+        on: c('Toggle button').t`On`,
+        off: c('Toggle button').t`Off`
+    };
+
+    return (
+        <span className="pm-toggle-label-text">
+            <Icon name={key} alt={I18N[key]} className="pm-toggle-label-img" />
+        </span>
+    );
+};
+
+const Toggle = ({ id, checked, onChange }) => {
+    const [value, setValue] = useState(checked);
+
+    /*
+        If you change the checked from the outside -> ex update the prop outside
+        we update the local state with the new value.
+    */
+    useEffect(() => {
+        setValue(checked);
+    }, [checked]);
+
+    const handleChange = () => {
+        const newVal = !value;
```

Comment:
It's better to get the value from the `event.target.checked` in case something interact with the `value` in the same component, see https://reactjs.org/docs/forms.html#controlled-components